universal-application-tool-0.0.1/app/controllers/dev/AzuriteFileStorageStrategy.java

```diff
@@ -16,19 +16,28 @@ public Result create(StoredFileRepository storedFileRepository, Request request)
     Optional<String> etag = request.queryString("etag");
     Optional<String> container = request.queryString("container");
     Optional<String> fileName = request.queryString("fileName");
-    Optional<String> userFileName = request.queryString("userFileName");
-    updateFileRecord(storedFileRepository, fileName.get());
+    Optional<String> originalFileName = request.queryString("originalFileName");
+    updateFileRecord(storedFileRepository, fileName.get(), originalFileName.get());
     String successMessage =
         String.format(
             "File successfully uploaded to Azure: container: %s, file name: %s, etag: %s, user"
                 + " file name: %s.",
-            container.get(), fileName.get(), etag.orElse(""), userFileName.get());
+            container.get(), fileName.get(), etag.orElse(""), originalFileName.get());
     return redirect(routes.FileUploadController.index().url()).flashing("success", successMessage);
   }
 
-  private void updateFileRecord(StoredFileRepository storedFileRepository, String key) {
+  private void updateFileRecord(
+      StoredFileRepository storedFileRepository, String key, String originalFileName) {
     StoredFile storedFile = new StoredFile();
     storedFile.setName(key);
+    storedFile.setOriginalFileName(getPrefixedOriginalFileName(key, originalFileName));
     storedFileRepository.insert(storedFile);
   }
+
+  /** Returns the file's original file name, in the format "dev/${filename}" */
+  private String getPrefixedOriginalFileName(String fileName, String originalFileName) {
```

Comment:
continuing from a previous comment thread:

> Added a comment. And yes, this will be used in production code too. The prefixedOriginalFileName will either be in the format "dev/${filename}" or "applicant-%d/program-%d/block-%s/${filename}" where $filename is the original name of the file. depending on whether the file is being uploaded from an application or the dev upload view.

question (blocking): How do the filenames end up with the URL path prefixed to the name?

suggestion (blocking): If the logic in this method will be needed in prod please extract it to its own class and add a unit test

Reply:
The prefixing happens here: https://github.com/seattle-uat/civiform/blob/4d1e90fddd3d6da2c4a249f4f78442d08f9088d3/universal-application-tool-0.0.1/app/views/applicant/ApplicantProgramBlockEditView.java#L128
and here for dev mode: https://github.com/seattle-uat/civiform/blob/main/universal-application-tool-0.0.1/app/controllers/dev/FileUploadController.java#L59

Reply:
Moved the method to the `FileNameFormatter` class and added a test.

Reply:
Great!

Suggestion (blocking): please extract the formatting of the filename into additional methods on `FileNameFormatter` for both dev and prod

e.g.
```java
public static String formatFileUploadQuestionFilename(ApplicantProgramBlockEditView.Params params) {
  return String.format(
            "applicant-%d/program-%d/block-%s",
            params.applicantId(), params.programId(), params.block().getId());
}
```